File: 938.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int rangeSumBST(TreeNode* root, int low, int high) {
        int totalSum = 0 ; 
        solve(root, low, high, totalSum);
        return totalSum; 
    }

    void solve(TreeNode* node, int& low, int& high, int& totalSum) { 
        if (node == NULL) return; 
        totalSum += node->val >= low && node->val <= high ? node->val: 0; 
        solve(node->left, low, high, totalSum);
        solve(node->right, low, high, totalSum); 

    }
};

File: 938.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solution {
    public int RangeSumBST(TreeNode root, int low, int high) {
        Queue<TreeNode> q = new Queue<TreeNode>(); 
        q.Enqueue(root);
        int total = 0; 
        while (q.Count != 0){
            TreeNode node = q.Dequeue();
            if (node.val >= low && node.val <= high) 
                total += node.val;
            if (node.right != null) 
                q.Enqueue(node.right);
            if (node.left != null)
                q.Enqueue(node.left); 
        }
        return total; 
    }
}
